### ISSUE TITLE
Uneven sampling if number of tfrecords > cycle_length

### ISSUE THREAD

**Issue by illuminascent** [google-research/ALBERT#177]
The input function defined in run_pretraining.py will make a repeating random sequence of tfrecord filenames and open limited amount of handles according to the sequence.

There is problem when number of files is greater than max handles, because a new set of handles will not be opened unless a certain amount of samples were read, this causes severe selection bias, and causes the loss to suddenly jump up or down during pretraining.

This behavior can be confirmed using a toy dataset. 10000 integers made using np.arange() were evenly divided into 10 parts and then shuffled within each part before being written into 10 separate tfrecords. Here's the value of each new sample drawn from the dataset at each training step(step in x axis and value in y).
![parallel_interleave_original](https://user-images.githubusercontent.com/44407415/75966878-5fd1fe80-5f0e-11ea-9508-cb8f70d53213.png)

I would suggest making the cycle_length to always match len(file_lists) and also move d=d.repeat() after the parallel interleave mapping. This solved the problem in my case at least.

Here's how it looks afterwards.
![parallel_interleave_cycle_extended_repetition_loop_changed](https://user-images.githubusercontent.com/44407415/75967692-893f5a00-5f0f-11ea-9150-2f67502817b9.png)
